### PR TITLE
Camera zones

### PR DESCRIPTION
**Pull request by jack27121**
I've added obj_stanncam_zone
which when the followed object is entered, constrains the camera to be within it's bounding box
Also important is i've added `zone_constrain_speed` to stanncam vars
It's used to lerp between normal and constrained positon. Setting it to 1 will be instantaneous, and 0.1 will be slower 

There's some different zone objects with different sides to constrain. They can be rotated 90degs  in the room. And ideally shouldn't overlap. As only one zone is in effect at a time

fixes https://github.com/jack27121/STANNcam/issues/39

https://github.com/jack27121/STANNcam/assets/46312671/28a9b2d6-83bd-460c-9727-004b22a48c7d

### CODE REVIEW

**Comment by bfrymire** [comment]
There's an issue in the example project with the look ahead debug box not drawing correctly.

https://github.com/jack27121/STANNcam/assets/20096702/2f94ef9f-b98a-4226-a581-1bfebd46f9a9

**Comment by jack27121** [comment]
Runner_fL7qYuHLr7 (2)
@bfrymire 
you can check again

File: scripts/stanncam/stanncam.gml [hunk 1]
```diff
@@ -571,22 +595,52 @@ function stanncam(_x=0, _y=0, _width=global.game_w, _height=global.game_h, _surf
...
 			
-			_new_x += constrain_offset_x;
-			_new_y += constrain_offset_y;
+			_new_x += __constrain_offset_x;
+			_new_y += __constrain_offset_y;
 		} else {
-			constrain_offset_x = 0;
-			constrain_offset_y = 0;
+			__constrain_offset_x = 0;
+			__constrain_offset_y = 0;
```
Comment (jack27121) [inline scripts/stanncam/stanncam.gml:641]: prepend __ since i think this variable isn't meant to be exposed

**Review by bfrymire — APPROVED** [review]

**Comment by jack27121** [comment]
GameMaker_kyXtC8eqcO
So i've now added a handful of zone objects, that can be placed and rotated (90deg) in the room to constrain the camera, they shouldn't overlap. If the camera is wider that the active zone then the camera will be constrained to the middle of the zone. Same with the height.

**Pull request merged**